packages/venia-concept/src/actions/cart/asyncActions.js

```diff
@@ -241,9 +241,18 @@ export const getCartDetails = (payload = {}) => {
 
             // for each item in the cart, look up its image in the cache
             // and merge it into the item object
+            // then assign its options from the totals subResource
             if (imageCache && Array.isArray(items) && items.length) {
+                const validTotals = totals && totals.items;
                 items.forEach(item => {
                     item.image = item.image || imageCache[item.sku] || {};
+
+                    item.options = validTotals
+                        ? JSON.parse(
+                              totals.items.find(t => t.item_id === item.item_id)
+                                  .options
+                          )
```

Comment:
There are three layers of Inception here (😛 ), let's reduce the mental overhead by using an `if` statement instead of a ternary (`?`). Something like:

```js
let options = [];
if (validTotals) {
  const matchingItem = totals.items.find(t => t.item_id === item.item_id);
  if (matchingItem && matchingItem.options) {
    options = JSON.parse(matchingItem.options);
  }
}
item.options = options;
```

Reply:
Haha good idea - ternaries can get a bit hard to read.